db/config.cc

```diff
@@ -931,6 +931,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , wasm_udf_total_fuel(this, "wasm_udf_total_fuel", value_status::Used, 100000000, "Wasmtime fuel a WASM UDF can consume before termination")
     , wasm_udf_memory_limit(this, "wasm_udf_memory_limit", value_status::Used, 2*1024*1024, "How much memory each WASM UDF can allocate at most")
     , relabel_config_file(this, "relabel_config_file", value_status::Used, "", "Optionally, read relabel config from file")
+    , object_storage_file(this, "object_storage_file", value_status::Used, "", "Optionally, read object-storage endpoints config from file")
```

Comment:
That's because this config file must contain secret keys for AWS access and my inner paranoid suggest that secrets are better be kept separately from generic configuration.

As far as "everything gets more complicated" is concerned. If keeping this data on db::config thing only one place would become simpler -- the sstables_manager() constructor won't need yet another argument and that's it. The rest would remain as it's here.